File: src/main/java/com/r0lex/eventmanager/service/PerformerService.java
package com.r0lex.eventmanager.service;

import com.r0lex.eventmanager.model.database.Performer;
import com.r0lex.eventmanager.repository.PerformerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class PerformerService {
    @Autowired
    private PerformerRepository performerRepository;

    public Mono<Performer> getPerformerById(String id) {
        return performerRepository.findById(id);
    }

    public Flux<Performer> getPerformers() {
        return performerRepository.findAll();
    }

    public void createPerformer(Performer performer) {
        performerRepository.save(performer).subscribe();
    }
}

File: src/app/service/performer.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs/Observable';
import {PerformerObject} from '../model/performerObject';

@Injectable()
export class PerformerService {

    constructor(private http: HttpClient) {
    }

    findAll(): Observable<PerformerObject[]> {
        return this.http.get<PerformerObject[]>('/performer/all');
    }

    findById(id: string): Observable<PerformerObject> {
        return this.http.get<PerformerObject>('/performer', {params: {id: id}});
    }

    save(performer: PerformerObject): void {
        this.http.post('/performer', performer).subscribe();
    }
}
